bin/coverage_doctest.py

```diff
@@ -65,7 +65,7 @@ def print_header(name, underline=None, color=None):
 
     print()
     if color:
-        print("%s%s%s" % (c_color % colors[color], name, c_normal))
+        print(f"{c_color % colors[color]}{name}{c_normal})")
```

Comment:
This sort of change is not needed. There is nothing wrong with `%` formatting.

New code can use f-strings but there is no need to remove `%` formatting in old code.

Reply:
https://pylint.pycqa.org/en/latest/user_guide/messages/convention/consider-using-f-string.html#consider-using-f-string-c0209

It is better to use only one style of string formatting across project, unless there is significant reason not to do it, in terms of ergonomics.

Reply:
I tend to agree. We are slowly replacing the one with the other. It doesn't have to be done all at once. But if we allow this sort of change it will encourage more and that will lead us to a desired state of having one type of format. It think the unused placeholders is a nice feature, but one might also argue that there is something nice about the explicit, local nature of the `%` formatting which clearly identifies what is to be included in the string at the point of printing so there is no need to look around for the variables that are being printed.

Reply:
The basic point I would say is just: if it ain't broke don't fix it.

Changing this one line of code is hardly likely to cause problems but it is also a waste of time. It would also be a waste of time to go round the whole codebase changing all the other occurrences and we would then have the potential for introducing bugs as well.

This one file has many other occurrences of % formatting. I don't suggest to fix them though: my suggested improvement for this file is just to delete it and add a note somewhere about how to use coverage and pytest.

Reply:
FWIW there are places in the codebase where % formatting is used on purpose, specifically the latex printer because there are too many curly braces that would need to be escaped. 

I also tend to agree with Oscar. While there's nothing wrong with this specific PR, changes that do nothing but shift code around don't really add anything and have the side effect of potentially introducing new bugs, and even if they don't, they waste maintainer time compared to changes that actually do something useful.

Reply:
> and even if they don't, they waste maintainer time compared to changes that actually do something useful.

Unfortunately, this looks like unfriendly direction to new contributors though.

It is more easy to get people involved with stylish changes than the changes to implement new algorithms or such.

Reply:
I tend to agree that we shouldn't have many 'dead code' lying around like `coverage_doctest`. 
I think that it was you @asmeurer who suggested to keep the old test suites

Reply:
I think pushing new contributors to making stylistic contributions is really the wrong way. It gives people the wrong impression about what we are actually looking for in open source, and starts people off on the wrong foot. Contributions to open source should be *useful* to the open source project. Every contribution takes up contributor time to review and test it, as well as adding inherent technical debt involved in maintaining any new features, and the risks involved if a new change introduces a bug. Changes should be worth that debt/risk. 

I am much, *much*, happier to see new contributors fix bugs. This is the sort of change that is actually meaningful to the project, and has an actual impact to the end users. Plus, if your goal of contributing to open source is to demonstrate technical competence for your open source resume, or just to learn more about programming, then fixing bugs will actually do this and improve your skills. Fixing stylistic things does the opposite. You learn nothing, and it has a negative effect to your open source resume for anyone looking who is actually technically competent enough to distinguish between a PR and *good* PR. 

(and just to be clear, I'm not trying to pick on @Maksim325. There are a lot of PRs made to SymPy and other projects that are like this. This is something I'd like to see changed more generally in open source)

Reply:
Although I'm not sure about whether this PR gives good stylish changes as well,
Unfortunately, I'd downvote SymPy specifically for the reason that its codebase is large, hard to understand, even compared to other open source libraries I see.

Reply:
> Fixing stylistic things does the opposite. You learn nothing, and it has a negative effect to your open source resume for anyone looking who is actually technically competent enough to distinguish between a PR and good PR.

Actually, bad code is often from some contributors who just tried to implement some feature, but are arrogant and doesn't want to follow the suggestions.

Reply:
I wouldn't describe the use of `%` formatting as bad code. There are some problems with `%` formatting that have been improved with the newer methods of string formatting but in many situations those problems don't apply. This particular file was added something like 15 years ago at which point str.format was just being added in Python 2.6 but f-strings were not added until much later.

It is not worth making any improvements let alone stylistic changes to this file when you can do:
```console
$ pytest --doctest-only --doctest-rst sympy/core/numbers.py --cov=sympy.core.numbers --cov-report=html
```
A better use of time would be making it so that the doctests pass under `--doctest-plus` and then replacing this script with something that just makes it a bit more convenient to generate the pytest command line shown above.

Reply:
> Така зміна не потрібна. У форматуванні немає нічого поганого `%`.
> 
> Новий код може використовувати f-рядки, але немає необхідності видаляти `%`форматування в старому коді.

Thank you, I understand and will avoid such unnecessary changes in the future

Reply:
> Краще використовувати лише один стиль форматування рядка в проекті, якщо немає серйозних причин не робити цього з точки зору ергономіки.
> 
> І я вважаю, що спільнота Python погоджується надавати перевагу `f-string`будь-якому іншому форматуванню, ніж навпаки.
> 
> https://pylint.pycqa.org/en/latest/user_guide/messages/convention/consider-using-f-string.html#consider-using-f-string-c0209
> 
> Наприклад, `%`синтаксис може стати неправильним, якщо врахувати можливість форматування кортежів
> 
> ```
> "%s" % (arg,)
> ```
> 
> і я також вважаю, що `f`рядки заохочують правильніший код, оскільки він видає помилки щодо невикористаних заповнювачів
> 
> ```
> "a {} b" # no error
> "a %s b" # no error
> f"a {x} b" # This errors for unused placeholder
> ```

If this is true, then I could have done worse for the project, so in the future I will try to avoid such changes. (It's just that this is my first open source project and I ran to change the code without getting used to it). Thank you for your time.

Reply:
I decided to look at the issues today and found out that I don't know algebra at all :( . Perhaps someone has some study materials I would be happy to discuss it